src/main/java/uk/gov/pay/api/service/GetPaymentService.java

```diff
@@ -1,49 +1,77 @@
 package uk.gov.pay.api.service;
 
+import uk.gov.pay.api.app.config.PublicApiConfig;
 import uk.gov.pay.api.auth.Account;
 import uk.gov.pay.api.exception.GetChargeException;
+import uk.gov.pay.api.ledger.service.LedgerUriGenerator;
+import uk.gov.pay.api.model.Charge;
 import uk.gov.pay.api.model.ChargeFromResponse;
+import uk.gov.pay.api.model.TransactionResponse;
 import uk.gov.pay.api.model.links.PaymentWithAllLinks;
 
 import javax.inject.Inject;
 import javax.ws.rs.client.Client;
 import javax.ws.rs.core.Response;
 import java.net.URI;
 
+import static org.apache.http.HttpStatus.SC_NOT_FOUND;
 import static org.apache.http.HttpStatus.SC_OK;
 
 public class GetPaymentService {
 
     private final Client client;
     private final PublicApiUriGenerator publicApiUriGenerator;
     private final ConnectorUriGenerator connectorUriGenerator;
+    private final LedgerUriGenerator ledgerUriGenerator;
+    private final PublicApiConfig config;
 
     @Inject
-    public GetPaymentService(Client client, PublicApiUriGenerator publicApiUriGenerator, ConnectorUriGenerator connectorUriGenerator) {
+    public GetPaymentService(Client client, PublicApiUriGenerator publicApiUriGenerator,
+                             ConnectorUriGenerator connectorUriGenerator, LedgerUriGenerator ledgerUriGenerator,
+                             PublicApiConfig config) {
         this.client = client;
         this.publicApiUriGenerator = publicApiUriGenerator;
         this.connectorUriGenerator = connectorUriGenerator;
+        this.ledgerUriGenerator = ledgerUriGenerator;
+        this.config = config;
     }
 
     public PaymentWithAllLinks getPayment(Account account, String paymentId) {
```

Comment:
It might be nice to move this towards a strategy pattern - especially as we introduce more feature flags etc to manage how we do various operations. But happy with this implementation for now

Reply:
Makes sense 👍 

Just drafted it quickly to have something working.